protocols/gossipsub/src/behaviour.rs

```diff
@@ -739,6 +750,14 @@ where
                 )
                 .is_err()
             {
+                self.failed_messages.entry(*peer_id).or_default().priority += 1;
```

Comment:
if a peer is in the mesh, and the policy is to drop messages, they will see an incomplete message stream which seems to me like it could lead to an inconsistent view of the universe - ie in a burst of messages, they'll get some, but not others.

For non-mesh-peers, this would be plugged with IHAVE, but not for mesh peers.. thoughts?